CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+# Contributing to this repository
+
+# Audience
+This set of contributing guidelines is for both external contributors, as well as Verne Global employees. There are number of references to using JIRA issue number in this document. If you are an external contributor and do not have access to the Verne Global JIRA instance, please ignore these references.
+
+# Table of Contents
+1. [Commits and pull requests](#commits-release-notes-pull-requests)
+    1. [Fixing a bug](#fixing-a-bug)
+    2. [Commits](#commits)
+    3. [Pull requests](#pull-requests)
+        1. [Reviewing pull requests](#reviewing-pull-requests)
+        2. [Merging pull requests](#merging-pull-requests)
+
+## Commits and pull requests
+### Fixing a bug
+
+1. Clone the repo, and commit changes to a new branch, named using the convention <JIRA-ISSUE-NUMBER>-description-of-change. e.g. ```HPCDIR-9999-add-contributing```
+2. Make pull requests against the master branch in the repo.
+3. Reference the JIRA issue in both the commit message and the pull request message.
+
+### Commits
+
+Please avoid the following in a commit:
+
+* Mixing whitespace changes with functional code changes.
+* Mixing two unrelated functional changes in the same commit.
+* Sending large new features in a single commit.
```

Comment:
Rob and I occasionally make a feature branch when we have large changes. In these situations, many small changes will be reviewed, and would then be merged into the feature branch. Once complete, this feature branch then gets it's own merge request, although it's mainly for interest as it would be too large to review by itself.

I only mention this as this is considered to be good practice, but does kind of break your rule a bit, so probably worth a mention

Reply:
This is in the commit section, so the idea is the same as above - to keep individual commits relatively small (sometimes unavoidable obviously), and then bundle related commits into a single PR/MR. 

Regarding the feature branches Rob and I chatted about this a bit yesterday (i.e. long running feature branch vs regular commits to master), and it seems like you guys have a need for the former, so we can edit the PR section as you see fit.